Formula/a/asdf.rb

```diff
@@ -16,38 +17,40 @@ class Asdf < Formula
     sha256 cellar: :any_skip_relocation, all: "a94dd362ce5c3a818f4fa56607eac3c616a3c1191e9f41480093bda2dc308af4"
   end
 
-  depends_on "autoconf"
-  depends_on "automake"
-  depends_on "coreutils"
-  depends_on "libtool"
-  depends_on "libyaml"
+  depends_on "go" => :build
+
   depends_on "openssl@3"
-  depends_on "readline"
-  depends_on "unixodbc"
 
   def install
-    bash_completion.install "completions/asdf.bash" => "asdf"
-    fish_completion.install "completions/asdf.fish"
-    zsh_completion.install "completions/_asdf"
-    libexec.install Dir["*"]
+    system("go", "build", *std_go_args(ldflags: "-s -X main.version=#{version}"), "./cmd/asdf")
+    original_stdout=$stdout
+
+    $stdout = File.new("asdf.bash", "w")
+    system(bin/"asdf", "completion", "bash")
+    bash_completion.install "asdf.bash" => "asdf"
+
+    $stdout = File.new("asdf.fish", "w")
+    system(bin/"asdf", "completion", "fish")
+    fish_completion.install "asdf.fish"
 
-    bin.write_exec_script libexec/"bin/asdf"
+    $stdout = File.new("_asdf", "w")
+    system(bin/"asdf", "completion", "zsh")
+    zsh_completion.install "_asdf"
+    $stdout = original_stdout
   end
 
   def caveats
     <<~EOS
-      To use asdf, add the following line (or equivalent) to your shell profile
-      e.g. ~/.profile or ~/.zshrc:
-        . #{opt_libexec}/asdf.sh
-      e.g. ~/.config/fish/config.fish
-        source #{opt_libexec}/asdf.fish
-      Restart your terminal for the settings to take effect.
+      To complete installation of asdf please follow the instructions on this page:
+
+      https://asdf-vm.com/guide/getting-started.html
```

Comment:
Duplicate the setup instructions for 5 different shells I just put a link to the documentation website which contains all the setup info.

Reply:
If it's not specific to Homebrew there shouldn't be a caveat at all.

Reply:
These instructions apply regardless of the installation method chosen. Whether brew, another package manager, or manual binary download additional steps will need to be followed to complete setup.

Reply:
Then they should not be a caveat. Those are only for places where `brew` differs from the norm.

Reply:
Caveat has been removed.